internal/vm/remotevm/vmsocket.go

```diff
@@ -0,0 +1,92 @@
+package remotevm
+
+import (
+	"context"
+	"io/ioutil"
+	"net"
+	"os"
+
+	"github.com/Microsoft/go-winio/pkg/guid"
+	"github.com/Microsoft/hcsshim/internal/vm"
+	"github.com/Microsoft/hcsshim/internal/vmservice"
+	"github.com/pkg/errors"
+)
+
+func (uvm *utilityVM) VMSocketListen(ctx context.Context, listenType vm.VMSocketType, connID interface{}) (_ net.Listener, err error) {
+	f, err := ioutil.TempFile("", "")
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create temp file for unix socket")
+	}
+
+	if err := f.Close(); err != nil {
+		return nil, errors.Wrap(err, "failed to close temp file")
+	}
+
+	if err := os.Remove(f.Name()); err != nil {
+		return nil, errors.Wrap(err, "failed to delete temp file to free up name")
+	}
```

Comment:
Is the point of this just to get a unique path to use?

Reply:
Yes, could make a comment about this.

Reply:
we don't care about putting these anywhere specific in case of issues where files aren't cleaned up or something? Maybe in the C:\ContainerPlatData dir like the container IO files or something?

Reply:
I don't think we should place them in a directory containerd maintains, but I'm not too worried about this in general. The way we setup the io channels is to just close the listener the second we accept a connection (which removes the socket file anyways).

Reply:
Could you explain why not?

Reply:
The socket file itself is removed the second you'd close the listener, but the connection is still valid. For a quick hacked up example.
```go
func echo(c net.Conn) {
	for {
		buf := make([]byte, 512)
		nr, err := c.Read(buf)
		if err != nil {
			log.Fatal(err)
		}

		data := buf[0:nr]
		fmt.Printf("Received: %v\n\n", string(data))
	}
}

func main() {
	sigChan := make(chan os.Signal, 1)
	signal.Notify(sigChan, syscall.SIGINT)

	l, err := net.Listen("unix", `/test.sock`)
	if err != nil {
		log.Fatal(err)
	}

	conn, err := l.Accept()
	if err != nil {
		log.Fatal(err)
	}

	// Socket file removed after this line, can still utilize the conn we have but the listeners
	// done for (can't accept any further conns).
	l.Close()

	go echo(conn)

	select {
	case <-sigChan:
		log.Print("Received signal, closing.")
	}
}
```
Where the listener we return in this func is eventually used is here: https://github.com/microsoft/hcsshim/blob/master/internal/gcs/iochannel.go#L14-L27

So I don't have many worries about socket files not being cleaned up.